docs/content/en/docs/yaml-crd-ref/taskdefinition.md

```diff
@@ -3,3 +3,276 @@ title: KeptnTaskDefinition
+
+    See [Create secret text](../implementing/tasks/#create-secret-text)
+    for details.
+
+## Usage
+
+A Task executes the TaskDefinition of a
+[KeptnApp](app.md) or [KeptnWorkload].
+The execution is done by spawning a Kubernetes
+[Job](https://kubernetes.io/docs/concepts/workloads/controllers/job/)
+to handle a single Task.
+In its state, it tracks the current status of this Kubernetes Job.
+
+The `function` is coded in JavaScript
+and executed in
+[Deno](https://deno.com/runtime),
+which is a lightweight runtime environment
+that executes in your namespace.
+Note that Deno has tighter restrictions
+for permissions and importing data
+so a script that works properly elsewhere
+may not function properly when run in Deno.
+
+A task can be executed either pre-deployment or post-deployment
+as specified in the `Deployment` resource;
+see
+[Pre- and post-deployment tasks](../implementing/integrate/#pre--and-post-deployment-checks)
+for details.
+Note that the annotation identifies the task by `name`.
+This means that you can modify the `function` code in the resource definition
+and the revised code is picked up without additional changes.
+
+## Examples
+
+### Example 1: inline script
+
+This example defines a full-fledged Deno script
+within the `KeptnTaskDefinition` YAML file:
+
+```yaml
+apiVersion: lifecycle.keptn.sh/v1alpha3
+kind: KeptnTaskDefinition
+metadata:
+  name: hello-keptn-inline
+spec:
+  function:
+    inline:
+      code: |
+        let text = Deno.env.get("DATA");
+        let data;
+        let name;
+        data = JSON.parse(text);
+
+        name = data.name
+        console.log("Hello, " + name + " new");
+```
+
+### Example 2: httpRef script
+
+This example fetches the Deno script from a remote webserver at runtime:
+
+```yaml
+apiVersion: lifecycle.keptn.sh/v1alpha3
+kind: KeptnTaskDefinition
+metadata:
+  name: hello-keptn-http
+spec:
+  function:
+    httpRef:
+      url: "https://www.example.com/yourscript.js"
+```
+
+For another example, see the
+[sample-app](https://github.com/keptn-sandbox/lifecycle-toolkit-examples/blob/main/sample-app/version-1/app-pre-deploy.yaml).
+
+See the
+[sample-app/version-1](https://github.com/keptn-sandbox/lifecycle-toolkit-examples/blob/main/sample-app/version-1/app-pre-deploy.yaml)
+PodtatoHead example for a more complete example.
+
+### Example 3: functionRef
+
+This example calls another defined task,
+illustrating how one `KeptnTaskDefinition` can build
+on top of other `KeptnTaskDefinition`s.
+In this case, it calls `slack-notification-dev`,
+passing `parameters` and `secureParameters` to that other task:
+
+```yaml
+apiVersion: lifecycle.keptn.sh/v1alpha3
+kind: KeptnTaskDefinition
+metadata:
+  name: slack-notification-dev
+spec:
+  function:
+    functionRef:
+      name: slack-notification
+    parameters:
+      map:
+        textMessage: "This is my configuration"
+    secureParameters:
+      secret: slack-token
+```
+
+### Example 4: ConfigMapRef
```

Comment:
can we add the config map too here? you can use this example https://github.com/keptn/lifecycle-toolkit/blob/main/operator/config/samples/function_execution/taskdefinition_function_configmap.yaml

Reply:
Added